frontend/app/src/components/dialogs/WatcherDialog.vue

```diff
@@ -0,0 +1,380 @@
+                label="Operation"
+                required
+              >
+              </v-select>
+            </v-col>
+            <v-col cols="4">
+              <v-text-field
+                v-model="watcherValue"
+                :label="watcherValueLabel"
+                hide-details
+                dense
+                outlined
+                suffix="%"
+              ></v-text-field>
+            </v-col>
+            <v-col cols="2" class="d-flex align-center justify-center">
+              <v-btn
+                small
+                icon
+                class="mt-1"
+                :disabled="watcherOperation === null || watcherValue === null"
+                @click="addWatcher()"
+              >
+                <v-icon>
+                  fa fa-plus
+                </v-icon>
+              </v-btn>
+            </v-col>
+          </v-row>
+        </v-row>
+      </v-card-text>
+
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn
+          depressed
+          color="primary"
+          class="watcher-dialog__buttons__close"
+          @click="cancel()"
+        >
+          Close
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator';
+import MessageDialog from '@/components/dialogs/MessageDialog.vue';
+import { Watcher, WatcherArgs } from '@/services/types-api';
+import { WatcherType } from '@/services/types-common';
+import { Message } from '@/store/store';
+
+@Component({
+  components: { MessageDialog }
+})
+export default class WatcherDialog extends Vue {
+  @Prop({ required: true })
+  title!: string;
+  @Prop({ required: true })
+  message!: string;
+  @Prop({ type: Boolean, required: true })
+  display!: boolean;
+
+  @Prop({ required: false, default: 'Watcher Value' })
+  watcherValueLabel!: string;
+  @Prop({ required: true, default: null })
+  watcherContentId!: string | number | null;
+  @Prop({ required: false, default: '' })
+  preselectWatcherType!: string;
+  @Prop({ required: false, type: Array, default: () => [] })
+  existingWatchers!: Watcher[];
+
+  watcherType: string | null = null;
+  watcherOperation: string | null = null;
+  watcherValue: string | null = null;
+
+  loadedWatchers: Watcher[] = [];
+  existingWatchersEdit: {
+    [identifier: string]: boolean;
+  } = {};
+
+  // mounted() {
+  //   this.watcherType = this.preselectWatcherType;
+  //   this.loadedWatchers = JSON.parse(JSON.stringify(this.existingWatchers)); // make a non-reactive copy
+
+  //   this.loadedWatchers.forEach(watcher => {
+  //     this.existingWatchersEdit[watcher.identifier] = false;
+  //   });
+  // }
+
+  @Watch('display')
+  onDialogToggle() {
+    if (this.display) {
+      this.watcherType = this.preselectWatcherType;
+      this.loadedWatchers = JSON.parse(JSON.stringify(this.existingWatchers)); // make a non-reactive copy
```

Comment:
isn't  `this.loadedWatchers = {...this.existingWatchers}` working the same way? you just want a copy right?

Reply:
I tried it but it didn't work, for some reason it was still reactive (changing loadedWatchers also changed existingWatchers).

Reply:
the objects themselves you mean?

Reply:
Yes.